agent/http_decode_test.go

```diff
@@ -0,0 +1,3304 @@
+		in:         []interface{}{`true`},
+		want:       true,
+		jsonFmtStr: "{" + tlsFields[0] + "}",
+		equalityFn: tlsEqFn,
+	},
+	{
+		desc:       "tlsSkipVerify: second set",
+		in:         []interface{}{`true`},
+		want:       true,
+		jsonFmtStr: "{" + tlsFields[1] + "}",
+		equalityFn: tlsEqFn,
+	},
+	{
+		desc:       "tlsSkipVerify: neither set",
+		in:         []interface{}{},
+		want:       false, // zero value
+		jsonFmtStr: "{}",
+		equalityFn: tlsEqFn,
+	},
+}
+
+// ServiceID: string
+func serviceIDEqFn(out interface{}, want interface{}) error {
+	var got interface{}
+	switch v := out.(type) {
+	case structs.CheckDefinition:
+		got = v.ServiceID
+	case *structs.CheckDefinition:
+		got = v.ServiceID
+	case structs.CheckType:
+		return nil // CheckType does not have a ServiceID field
+	case *structs.CheckType:
+		return nil // CheckType does not have a ServiceID field
+	case structs.HealthCheckDefinition:
+		return nil // HealthCheckDefinition does not have a ServiceID field
+	case *structs.HealthCheckDefinition:
+		return nil // HealthCheckDefinition does not have a ServiceID field
+	default:
+		panic(fmt.Sprintf("unexpected type %T", out))
+	}
+	if got != want {
+		return fmt.Errorf("expected ServiceID to be %s, got %s", want, got)
+	}
+	return nil
+}
+
+var serviceIDFields = []string{`"ServiceID": %s`, `"service_id": %s`}
+var translateServiceIDTCs = []translateKeyTestCase{
+	{
+		desc:       "serviceID: both set",
+		in:         []interface{}{`"id-1"`, `"id-2"`},
+		want:       "id-1",
+		jsonFmtStr: "{" + strings.Join(serviceIDFields, ",") + "}",
+		equalityFn: serviceIDEqFn,
+	},
+	{
+		desc:       "serviceID: first set",
+		in:         []interface{}{`"id-1"`},
+		want:       "id-1",
+		jsonFmtStr: "{" + serviceIDFields[0] + "}",
+		equalityFn: serviceIDEqFn,
+	},
+	{
+		desc:       "serviceID: second set",
+		in:         []interface{}{`"id-2"`},
+		want:       "id-2",
+		jsonFmtStr: "{" + serviceIDFields[1] + "}",
+		equalityFn: serviceIDEqFn,
+	},
+	{
+		desc:       "serviceID: neither set",
+		in:         []interface{}{},
+		want:       "", // zero value
+		jsonFmtStr: "{}",
+		equalityFn: serviceIDEqFn,
+	},
+}
+
+// ==================================
+// $GOPATH/github.com/hashicorp/consul/agent/acl_endpoint.go:
+//   327  	s.parseToken(req, &args.Token)
+//   328
+//   329: 	if err := decodeBody(req, &args.Policy, fixTimeAndHashFields); err != nil {
+//   330  		return nil, BadRequestError{Reason: fmt.Sprintf("Policy decoding failed: %v", err)}
+//   331  	}
+// ==================================
+
+// ACLPolicySetRequest:
+// Policy	structs.ACLPolicy
+//     ID	string
+//     Name	string
+//     Description	string
+//     Rules	string
+//     Syntax	acl.SyntaxVersion
+//     Datacenters	[]string
+//     Hash	[]uint8
+//     RaftIndex	structs.RaftIndex
+//         CreateIndex	uint64
+//         ModifyIndex	uint64
+// Datacenter	string
+// WriteRequest	structs.WriteRequest
+//     Token	string
```

Comment:
These comments are _so_ helpful. Thanks for the review UX consideration!

Reply:
:tada: ! thanks for the review !